client/src/js/services/grid/GridColumns.js

```diff
@@ -53,8 +53,10 @@ function GridColumnService(uiGridConstants, AppCache, Modal, util) {
     });
 
     // if there is a cache defined, load it into the current view
-    if (Object.keys(cache).length > 0) {
-      this.setVisibleColumns(cache);
+    if(cache){
+      if (Object.keys(cache).length > 0) {
```

Comment:
Just a note, this could be
```js
if (cache && Object.keys(cache).length > 0) {
```
to avoid the [pyramid of doom](https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming)).